sbin/utils.py

```diff
@@ -0,0 +1,29 @@
+import re
+import subprocess
+
+possible_types = ["feat", "fix", "refactor", "docs", "chore", "ci"]
+
+types = "|".join(possible_types)
+
+re_conventional_commit_header = re.compile(fr"^({types})(?:\(([^\)]+)\))?(!?): (.*)$")
+
+
+def conventional_commit_parse(message: str):
+    match = re.fullmatch(re_conventional_commit_header, message)
+
+    if match is None:
+        raise Exception()
+
+    type, scope, breaking, header = match.groups()
+
+    return {
+        "type": type,
+        "scope": scope,
+        "breaking": breaking == "!",
+        "description": header,
+    }
+
+
+def shell(cmd: str) -> str:
+    """ Call shell command and return the result """
+    return subprocess.check_output(cmd, shell=True, encoding="utf-8").strip()
```

Comment:
If everything is replaced with Python (cfr release script as well), then why not go the extra mile of replacing dependency on a shell?

Reply:
Done